lib/IMAP/PreviewEnhancer.php

```diff
@@ -34,22 +35,32 @@ class PreviewEnhancer {
 	/** @var LoggerInterface */
 	private $logger;
 
+	/** @var AvatarService */
+	private $avatarService;
+
+	/** @var string|null */
+	private $userId;
+
 	public function __construct(IMAPClientFactory $clientFactory,
 		ImapMapper $imapMapper,
 		DbMapper $dbMapper,
-		LoggerInterface $logger) {
+		LoggerInterface $logger,
+		AvatarService $avatarService,
+		string $userId) {
 		$this->clientFactory = $clientFactory;
 		$this->imapMapper = $imapMapper;
 		$this->mapper = $dbMapper;
 		$this->logger = $logger;
+		$this->avatarService = $avatarService;
+		$this->userId = $userId;
```

Comment:
nitpick: I see injection of data as a code smell. The only exception for a valid userId injection is a controller.

Would it be possible to move the transport of the user context (userId) to the method parameters?